transmart-rest-api/src/main/groovy/org/transmartproject/rest/serialization/tabular/TabularResultSPSSSerializer.groovy

```diff
@@ -105,25 +109,30 @@ class TabularResultSPSSSerializer implements TabularResultSerializer {
     }
 
     static writeValues(TabularResult tabularResult, OutputStream outputStream) {
-        CSVWriter csvWriter = new CSVWriter(new OutputStreamWriter(outputStream, 'utf-8'), COLUMN_SEPARATOR, CSVWriter.NO_QUOTE_CHARACTER)
+        CSVWriter csvWriter = new CSVWriter(
+                new BufferedWriter(
+                        new OutputStreamWriter(outputStream, 'utf-8'),
+                        // large 32k chars buffer to reduce overhead
+                        32*1024),
+                COLUMN_SEPARATOR, CSVWriter.NO_QUOTE_CHARACTER)
         List<DataColumn> columns = tabularResult.indicesList
         csvWriter.writeNext(columns.collect { toSpssLabel(it.label) } as String[])
-        tabularResult.rows.each { DataRow row ->
-            List valuesRow = columns.collect { DataColumn column -> row[column] }
+        tabularResult.rows.forEachRemaining({DataRow row ->
+            List<Object> valuesRow = columns.stream().map({DataColumn column -> row[column]}).collect(Collectors.toList())
             csvWriter.writeNext(formatRowValues(valuesRow))
-        }
+        })
         csvWriter.flush()
     }
 
     private static String[] formatRowValues(List<Object> valuesRow) {
-        valuesRow.collect { value ->
+        valuesRow.stream().map({value ->
```

Comment:
What does it solve?

Reply:
I'm trying to avoid dynamic closures in the critical path. I'm not sure if it will give much performance gain.